app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt

```diff
@@ -118,6 +118,9 @@ class CtaViewModel @Inject constructor(
                 pixel.fire(it, cta.pixelShownParameters())
             }
         }
+        if (cta is OnboardingDaxDialogCta && cta.markAsReadOnShow) {
```

Comment:
Do you actually need `markAsReadOnShow`? Since `markAsReadOnShow` is only true for `DaxEndCta`, can’t you do something like this instead?:
`if (cta is DaxEndCta) ...`

Reply:
I did it like this for it to scale? This way, if we want to add more dialogs which will be marked as read when shown, we'll just need to set the variable to true and it will work without any extra implementation.

Reply:
Okay, we can leave it in then 👍